pymc3/tests/test_theanof.py

```diff
@@ -1,9 +1,61 @@
 import collections
-
 import pytest
-from theano import theano
+from itertools import product
+
+from theano import theano, tensor as tt
+import numpy as np
+
+from pymc3.theanof import set_theano_conf, take_along_axis, _conversion_map
+from pymc3.vartypes import int_types
+
+
+FLOATX = str(theano.config.floatX)
+INTX = str(_conversion_map[FLOATX])
+
+
+def _make_along_axis_idx(arr_shape, indices, axis):
+    # compute dimensions to iterate over
+    if str(indices.dtype) not in int_types:
+        raise IndexError('`indices` must be an integer array')
+    shape_ones = (1,) * indices.ndim
+    dest_dims = list(range(axis)) + [None] + list(range(axis+1, indices.ndim))
 
-from pymc3.theanof import set_theano_conf
+    # build a fancy index, consisting of orthogonal aranges, with the
+    # requested index inserted at the right location
+    fancy_index = []
+    for dim, n in zip(dest_dims, arr_shape):
+        if dim is None:
+            fancy_index.append(indices)
+        else:
+            ind_shape = shape_ones[:dim] + (-1,) + shape_ones[dim+1:]
+            fancy_index.append(np.arange(n).reshape(ind_shape))
+
+    return tuple(fancy_index)
+
+
+if hasattr(np, "take_along_axis"):
```

Comment:
```suggestion
# np.take_along_axis added in 1.18.0, so 
# this can be removed once this version of numpy is required
if hasattr(np, "take_along_axis"):
```